### PR TITLE
[WIP] Replacing wreck with fetch

### PR DESCRIPTION
**Pull request by dignifiedquire**
As discussed in #353 this can help us reduce the size of the bundle. 
- [x] Reenable browser tests on CI
- [x] Get browser tests to run without memory flag
- [x] Tests passing in node
- [x] Tests passing in phantomjs
- [x] Tests passing in Chrome
- [ ] Figure out `log` tests (passing before, but shouldn't have)

### CODE REVIEW

**Comment by diasdavid** [comment]
Circle is still not happy, seems that you managed to reduce the memory fingerprint, but not totally:

```
<--- Last few GCs --->

   97037 ms: Mark-sweep 1346.4 (1422.4) -> 1346.0 (1423.4) MB, 172.9 / 0 ms (+ 5.6 ms in 42 steps since start of marking, biggest step 1.7 ms) [allocation failure] [GC in old space requested].
   97220 ms: Mark-sweep 1346.0 (1423.4) -> 1346.0 (1424.4) MB, 182.1 / 0 ms [allocation failure] [GC in old space requested].
   97444 ms: Mark-sweep 1346.0 (1424.4) -> 1346.0 (1418.4) MB, 224.2 / 0 ms [last resort gc].
   97641 ms: Mark-sweep 1346.0 (1418.4) -> 1345.9 (1414.4) MB, 197.2 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x16cfd8ac9e31 <JS Object>
    1: slowToString [buffer.js:467] [pc=0x35bbcf78bb75] (this=0x93d6d983831 <an Uint8Array with map 0x54321fb34a1>,encoding=0x93d6d983711 <String[5]: utf-8>,start=0,end=22984396)
    2: arguments adaptor frame: 1->3
    3: toString [buffer.js:~495] [pc=0x35bbcf768f03] (this=0x93d6d983831 <an Uint8Array with map 0x54321fb34a1>)
    4: arguments adaptor frame: 1->0
    5: inlineMap [/home/ubuntu/...
```

File: test/interface-ipfs-core/log.spec.js [hunk 1]
```diff
@@ -10,7 +10,9 @@ const FactoryClient = require('../factory/factory-client')
 const isPhantom = !isNode && typeof navigator !== 'undefined' && navigator.userAgent.match(/PhantomJS/)
 
 if (!isPhantom) {
-  describe('.log', () => {
+  // How did this work before
+  // our polyfill didn't have real streaming support
+  describe.skip('.log', () => {
```
Comment (diasdavid) [inline test/interface-ipfs-core/log.spec.js]: Polyfill? That is why we have the `if(!isPhantom)`

**Comment by dignifiedquire** [comment]
Real streaming is working in Node.js and Chrome. For other browsers I need to implement a fallback onto xhr based streaming.

**Comment by dignifiedquire** [comment]
Replaced by #414

**Pull request closed without merging**